composer/utils/reproducibility.py

```diff
@@ -86,7 +87,9 @@ def get_random_seed() -> int:
     Returns:
         int: A random seed.
     """
-    seed = int(torch.empty((), dtype=torch.int64).random_(to=2**32).item())
+    rng = random.Random(int(time.time_ns()))  # get a new RNG does not respect the current seed
+    seed = rng.randint(0, 2**32 - 1)
+    assert seed >= 0 and seed < 2**32, "seed should be on this range"
```

Comment:
Can we make 2**32 a constant called `MAX_ALLOWED_SEED` since it's used in trainer.py as well?